src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruitName() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+        private static final Map<String, Operation> operationMap;
+        private String code;
+
+        static {
+            operationMap = new HashMap<>();
+            operationMap.put("b", FruitTransaction.Operation.BALANCE);
+            operationMap.put("s", FruitTransaction.Operation.SUPPLY);
+            operationMap.put("p", FruitTransaction.Operation.PURCHASE);
+            operationMap.put("r", FruitTransaction.Operation.RETURN);
+        }
```

Comment:
You don`t need static block, save it to constant using map.of(), you can place it in class, it will be better or create own class with this map like storage